app/models/form_response.rb

```diff
@@ -1,4 +1,7 @@
 # frozen_string_literal: true
 
 class FormResponse < ApplicationRecord
+  JSON_SCHEMA = Rails.root.join("config/schemas/form_response.json").to_s
+
+  validates :form_response_v2, presence: true, json: { message: ->(errors) { errors }, schema: JSON_SCHEMA }
```

Comment:
I wonder if we can find a way to log validation errors here, but not prevent users from submitting data, until we're sure the schema is complete, to be sure we'll not prevent users from submitting valid data.

Reply:
I don't think that's how validations are designed to work. I'm using `create!` rather than `create` so an exception is raised, which I've assumed will be logged in Sentry.

I think this would work though:

```
fr = FormResponse.new(form_response: session)
if fr.invalid? 
   # Log error somehow
end

fr.save(validate: false)
```

Reply:
Or

```
    unless smoke_tester?
      begin
        FormResponse.create!(form_response: session.to_hash)
      rescue ActiveRecord::RecordInvalid => e
        logger.info do
          {
            validation_error: { text: e.message },
          }.to_json

        FormResponse.create(form_response: session.to_hash, validate: false)
      end
    end
```

Reply:
Sorry forgot to add a comment on this one. We discussed offline - iirc the plan is to send an error to sentry if the data isn't valid against the schema, but not prevent users submitting the form, and keeping the existing model validations in place if possible.